cmake/check_compiler_flag.cmake

```diff
@@ -54,3 +54,43 @@ FUNCTION(MY_CHECK_AND_SET_COMPILER_FLAG flag_to_set)
     ENDIF()
   ENDFOREACH()
 ENDFUNCTION()
+
+FUNCTION(MY_CHECK_AND_SET_COMPILER_FLAG_CC flag_to_set)
+  # At the moment this is gcc-only.
+  # Let's avoid expensive compiler tests on Windows:
+  IF(WIN32)
```

Comment:
The comment is wrong, it is not `gcc-only`. It is clang and gcc, on Unixes only. 

Also, IF(WIN32) needs to be fixed one day, because there is a clang on Windows, which supports a plenty of options that are tested. But you do not need to fix it in your patch. This is a sidenote for myself.